core/embed/trezorhal/stm32u5/trustzone.c

```diff
@@ -1,200 +1,114 @@
-  __HAL_RCC_GTZC1_CLK_ENABLE();
-  __HAL_RCC_GTZC2_CLK_ENABLE();
-
-  /* -------------------------------------------------------------------------*/
-  /*                   Memory isolation configuration                         */
-  /* Initializes the memory that secure application books for non secure      */
-  /* -------------------------------------------------------------------------*/
-
-  /* -------------------------------------------------------------------------*/
-  /* Internal RAM :                                                  */
-  /* The booking is done through GTZC MPCBB.                         */
-  /* Internal SRAMs are secured by default and configured by block   */
-  /* of 512 bytes.                                                   */
-
-  MPCBB_desc.SecureRWIllegalMode = GTZC_MPCBB_SRWILADIS_DISABLE;
-  MPCBB_desc.InvertSecureState = GTZC_MPCBB_INVSECSTATE_NOT_INVERTED;
-  MPCBB_desc.AttributeConfig.MPCBB_LockConfig_array[0] =
-      0x00000000U; /* Unlocked configuration */
-
-  for (index = 0; index < 52; index++) {
-    MPCBB_desc.AttributeConfig.MPCBB_SecConfig_array[index] = 0xFFFFFFFFU;
-    MPCBB_desc.AttributeConfig.MPCBB_PrivConfig_array[index] = 0x00000000U;
+// Configure SRAM security
+static void trustzone_configure_sram() {
+  MPCBB_ConfigTypeDef mpcbb = {};
+
+  // No exceptions on illegal access
+  mpcbb.SecureRWIllegalMode = GTZC_MPCBB_SRWILADIS_DISABLE;
+  // Settings of SRAM clock in RCC is secure
+  mpcbb.InvertSecureState = GTZC_MPCBB_INVSECSTATE_NOT_INVERTED;
+  // Set configuration as unlocked
+  mpcbb.AttributeConfig.MPCBB_LockConfig_array[0] = 0x00000000U;
+
+  // Set all blocks secured & unprivileged
+  for (int index = 0; index < 52; index++) {
+    mpcbb.AttributeConfig.MPCBB_SecConfig_array[index] = 0xFFFFFFFFU;
+    mpcbb.AttributeConfig.MPCBB_PrivConfig_array[index] = 0x00000000U;
   }
 
-  HAL_GTZC_MPCBB_ConfigMem(SRAM1_BASE, &MPCBB_desc);
-  HAL_GTZC_MPCBB_ConfigMem(SRAM2_BASE, &MPCBB_desc);
-  HAL_GTZC_MPCBB_ConfigMem(SRAM3_BASE, &MPCBB_desc);
-  HAL_GTZC_MPCBB_ConfigMem(SRAM4_BASE, &MPCBB_desc);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM1_BASE, &mpcbb);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM2_BASE, &mpcbb);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM3_BASE, &mpcbb);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM4_BASE, &mpcbb);
 #if defined STM32U5A9xx | defined STM32U5G9xx
-  HAL_GTZC_MPCBB_ConfigMem(SRAM5_BASE, &MPCBB_desc);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM5_BASE, &mpcbb);
 #endif
 #if defined STM32U5G9xx
-  HAL_GTZC_MPCBB_ConfigMem(SRAM6_BASE, &MPCBB_desc);
+  HAL_GTZC_MPCBB_ConfigMem(SRAM6_BASE, &mpcbb);
 #endif
+}
 
-  /* -------------------------------------------------------------------------*/
-  /* Internal Flash */
-  /* The booking is done in both IDAU/SAU and FLASH interface */
-
-  /* Flash memory is secured by default and modified with Option Byte Loading */
-  /* Insure SECWM2_PSTRT > SECWM2_PEND in order to have all Bank2 non-secure  */
+// Configure FLASH security
+static void trustzone_configure_flash() {
```

Comment:
static void trustzone_configure_flash(**void**) , applies to other functions as well